pkg/assessor/passwd/passwd.go

```diff
@@ -7,27 +7,30 @@ import (
 	"os"
 	"strings"
 
-	deckodertypes "github.com/goodwithtech/deckoder/types"
-
 	"github.com/Portshift/dockle/pkg/log"
-
 	"github.com/Portshift/dockle/pkg/types"
 )
 
 type PasswdAssessor struct{}
 
-func (a PasswdAssessor) Assess(fileMap deckodertypes.FileMap) ([]*types.Assessment, error) {
+func (a PasswdAssessor) Assess(imageData *types.ImageData) ([]*types.Assessment, error) {
 	log.Logger.Debug("Start scan : password files")
 
 	var existFile bool
 	assesses := []*types.Assessment{}
 	for _, filename := range a.RequiredFiles() {
-		file, ok := fileMap[filename]
+		file, ok := imageData.FileMap[filename]
 		if !ok {
 			continue
 		}
 		existFile = true
-		scanner := bufio.NewScanner(bytes.NewBuffer(file.Body))
+
+		content, err := file.ReadContentAndClose(imageData.Image)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read content and close: %w", err)
+		}
+
+		scanner := bufio.NewScanner(bytes.NewBuffer(content))
```

Comment:
why about checking for `scanner.Err()`?
```
	if err = scanner.Err(); err != nil {
		return xxxxx
	}
```

Reply:
Tis is old part of code I've just replaced the file.Body to content, but I am checking.

Reply:
I've added missing error check.

Reply:
`scanner.Err()` should be called after the `for scanner.Scan() {` loop